modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/SyncmlGenerator.java

```diff
@@ -42,7 +42,7 @@
 
     private static Log log = LogFactory.getLog(SyncmlGenerator.class);
 
-    public String generatePayload(SyncmlDocument syncmlDocument) throws WindowsOperationException {
+    public String generatePayload(SyncmlDocument syncmlDocument) throws SyncmlOperationException {
         Document doc = generateDocument();
         Element rootElement = createRootElement(doc);
         SyncmlHeader header = syncmlDocument.getHeader();
```

Comment:
As a public method exposed to outside, better checking the validity of the argument object (for ex: null or not) before calling any related functions of it and throw a runtime exception like "IllegalArgumentException".